pkg/operator/ceph/object/controller_test.go

```diff
@@ -377,9 +410,17 @@ func TestCephObjectStoreController(t *testing.T) {
 		s := scheme.Scheme
 		s.AddKnownTypes(cephv1.SchemeGroupVersion, &cephv1.CephObjectStore{})
 		s.AddKnownTypes(cephv1.SchemeGroupVersion, &cephv1.CephCluster{})
+		s.AddKnownTypes(v1.SchemeGroupVersion, &v1.Secret{})
 
 		// Create a fake client to mock API calls.
 		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+
+		// overwrite adminops context func
+		newMultisiteAdminOpsCtxFunc = mockMultisiteAdminOpsCtxFunc
+		defer func() {
+			mockMultisiteAdminOpsCtxFunc = newMultisiteAdminOpsCtxFunc
+		}()
```

Comment:
This doesn't effectively do anything on deferral. To reset the function back to its default, an intermediate storage var is necessary.

```suggestion
		oldNewMultisiteAdminOpsCtxFunc := newMultisiteAdminOpsCtxFunc
		newMultisiteAdminOpsCtxFunc = mockMultisiteAdminOpsCtxFunc
		defer func() {
			newMultisiteAdminOpsCtxFunc = oldNewMultisiteAdminOpsCtxFunc
		}()
```